src/uct/sm/base/sm_iface.c

```diff
@@ -38,17 +50,100 @@ uct_sm_base_query_tl_devices(uct_md_h md, uct_tl_device_resource_t **tl_devices_
                                       num_tl_devices_p);
 }
 
-ucs_status_t uct_sm_iface_get_device_address(uct_iface_t *tl_iface,
-                                             uct_device_addr_t *addr)
+
+/* read boot_id GUID or use machine_guid */
+static uint64_t uct_sm_iface_get_system_id()
+{
+    static ucs_init_once_t init_once = UCS_INIT_ONCE_INITIALIZER;
+    static uint64_t system_id        = 0;
+    const char *boot_id_filename     = "/proc/sys/kernel/random/boot_id";
+    const char *guid_scan_fmt        = "%x-%4hx-%4hx-%4hx-%2hhx%2hhx%2hhx%2hhx%2hhx%2hhx";
+    char guid_str[256];
+    ssize_t size;
+    int res;
+
+    union {
+        struct {
+            uint32_t v1;
+            uint16_t v2;
+            uint16_t v3;
+            uint16_t v4;
+            uint8_t  v5[6];
+        };
+        struct {
+            uint64_t low_dword;
+            uint64_t hi_dword;
+        };
+    } UCS_S_PACKED guid;
+
+    UCS_INIT_ONCE(&init_once) {
```

Comment:
add ucs function to read 128-bit boot id (or error). no need to use union, better use bit shift
ucs_status_t ucs_sys_get_boot_id(uint64_t *high, uint64_t *low)